R/pkg/inst/tests/testthat/test_sparkSQL.R

```diff
@@ -1137,6 +1137,13 @@ test_that("string operators", {
   expect_equal(count(where(df, like(df$name, "A%"))), 1)
   expect_equal(count(where(df, startsWith(df$name, "A"))), 1)
   expect_equal(first(select(df, substr(df$name, 1, 2)))[[1]], "Mi")
+  if (as.numeric(R.version$major) >= 3 && as.numeric(R.version$minor) >= 3) {
```

Comment:
actually, I think you should only put line 1144 and 1145 into the version check. startsWith(character) and endsWith(character) are not in R 3.2~ that's why the tests fail in Jenkins. The other with DataFrame/column should work though and we should test for it.